lib/matplotlib/tests/test_axes.py

```diff
@@ -2437,16 +2437,17 @@ def test_stairs_update(fig_test, fig_ref):
 
 
 @check_figures_equal(extensions=['png'])
-def test_stairs_baseline_0(fig_test, fig_ref):
-    # Test
-    test_ax = fig_test.add_subplot()
-    test_ax.stairs([5, 6, 7], baseline=None)
+def test_stairs_baseline_None(fig_test, fig_ref):
+    x = np.array([0, 2, 3, 5, 10])
+    y = np.array([1.148, 1.231, 1.248, 1.25])
+
+    test_axes = fig_test.add_subplot()
+    test_axes.stairs(y, x, baseline=None)
 
-    # Ref
-    ref_ax = fig_ref.add_subplot()
     style = {'solid_joinstyle': 'miter', 'solid_capstyle': 'butt'}
-    ref_ax.plot(range(4), [5, 6, 7, 7], drawstyle='steps-post', **style)
-    ref_ax.set_ylim(0, None)
```

Comment:
For next reviewer: removing `set_ylim` is the critical change in the test function.